File: examples/rust/src/b/higher_order_function.rs
use crate::kalai::kalai;
use crate::kalai::kalai::*;
pub fn main() {
    let _args: std::vec::Vec<String> = std::env::args().collect();
    {
        {
            let x: std::vec::Vec<i64> = {
                let mut tmp1: std::vec::Vec<i64> = std::vec::Vec::new();
                tmp1.push(1i64);
                tmp1.push(2i64);
                tmp1.push(3i64);
                tmp1.push(4i64);
                tmp1.push(5i64);
                tmp1
            };
            println!(
                "{} {}",
                String::from("HELLO***"),
                x.clone()
                    .into_iter()
                    .map(|y: i64| {
                        return (y + 1i64);
                    })
                    .next()
                    .unwrap()
                    .clone()
            );
        }
        {
            let x: std::vec::Vec<i64> = {
                let mut tmp2: std::vec::Vec<i64> = std::vec::Vec::new();
                tmp2.push(1i64);
                tmp2.push(2i64);
                tmp2.push(3i64);
                tmp2.push(4i64);
                tmp2.push(5i64);
                tmp2
            };
            println!(
                "{} {}",
                String::from("HELLO***"),
                x.clone()
                    .into_iter()
                    .map(|a| { (a + 1i64) })
                    .next()
                    .unwrap()
                    .clone()
            );
        }
        {
            let y: std::vec::Vec<i64> = {
                let mut tmp3: std::vec::Vec<i64> = std::vec::Vec::new();
                tmp3.push(1i64);
                tmp3.push(2i64);
                tmp3.push(3i64);
                tmp3.push(4i64);
                tmp3.push(5i64);
                tmp3
            };
            let z: i64 = y
                .clone()
                .into_iter()
                .reduce(|a, b| {
                    return (a + b);
                })
                .unwrap();
            let z2: String = y.clone().into_iter().fold(String::from(""), |a, b| {
                return format!("{}{}", a, b);
            });
            {
                println!("{} {}", String::from("z ="), z);
                println!("{} {}", String::from("z2 ="), z2);
            }
        }
        {
            let y: std::vec::Vec<i64> = {
                let mut tmp4: std::vec::Vec<i64> = std::vec::Vec::new();
                tmp4.push(1i64);
                tmp4.push(2i64);
                tmp4.push(3i64);
                tmp4.push(4i64);
                tmp4.push(5i64);
                tmp4
            };
            let z: i64 = y.clone().into_iter().reduce(|a, b| (a + b)).unwrap();
            let z2: String = y.clone().into_iter().fold(String::from(""), |a, b| {
                return format!("{}{}", a, b);
            });
            {
                println!("{} {}", String::from("z ="), z);
                println!("{} {}", String::from("z2 ="), z2);
            }
        }
    }
}

File: examples/java/src/b/HigherOrderFunction.java
package b;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.ArrayList;
import java.util.stream.Collectors;
public class HigherOrderFunction {
public static final void main(String[] _args) {
{
ArrayList<Long> tmp1 = new ArrayList<Long>();
tmp1.add(1L);
tmp1.add(2L);
tmp1.add(3L);
tmp1.add(4L);
tmp1.add(5L);
final ArrayList<Long> x = tmp1;
System.out.println(("" + "HELLO***" + x.stream().map((y) -> {
return (y + 1L);
}).findFirst().get()));
}
{
ArrayList<Long> tmp2 = new ArrayList<Long>();
tmp2.add(1L);
tmp2.add(2L);
tmp2.add(3L);
tmp2.add(4L);
tmp2.add(5L);
final ArrayList<Long> x = tmp2;
System.out.println(("" + "HELLO***" + x.stream().map((a) -> {
return (a + 1L);
}).findFirst().get()));
}
{
ArrayList<Long> tmp3 = new ArrayList<Long>();
tmp3.add(1L);
tmp3.add(2L);
tmp3.add(3L);
tmp3.add(4L);
tmp3.add(5L);
final ArrayList<Long> y = tmp3;
final long z = y.stream().reduce((a, b) -> {
return (a + b);
}).get();
final String z2 = kalai.Kalai.foldLeft(y, "", (a, b) -> {
return ("" + a + b);
});
{
System.out.println(("" + "z =" + z));
System.out.println(("" + "z2 =" + z2));
}
}
{
ArrayList<Long> tmp4 = new ArrayList<Long>();
tmp4.add(1L);
tmp4.add(2L);
tmp4.add(3L);
tmp4.add(4L);
tmp4.add(5L);
final ArrayList<Long> y = tmp4;
final long z = y.stream().reduce((a, b) -> {
return (a + b);
}).get();
final String z2 = kalai.Kalai.foldLeft(y, "", (a, b) -> {
return ("" + a + b);
});
{
System.out.println(("" + "z =" + z));
System.out.println(("" + "z2 =" + z2));
}
}
}
}
